lib/stripe.js

```diff
@@ -251,6 +261,50 @@ Stripe.prototype = {
       this[utils.pascalToCamelCase(name)] = new resources[name](this);
     }
   },
+
+  _getPropsFromConfig(config) {
+    // config can be an object or a string
+    const isString = typeof config === 'string';
+    const isObject = config === Object(config) && !Array.isArray(config);
+
+    if (!isObject && !isString) {
+      throw new Error('Config must either be an object or a string');
+    }
```

Comment:
nit: What do you think about checking *all four* cases (undefined, string, object, other) in a single place, and producing the canonical form {version: string} immediately, then performing the version string validation along with the other validation?

Reply:
Damnit that's so much better

Reply:
I rejigged a bit and think this "return fast and early" strategy works, let me know what you think.